pkg/cmd/flagtypes/addr.go

```diff
@@ -55,26 +57,59 @@ func (a *Addr) String() string {
 
 // Set attempts to set a string value to an address
 func (a *Addr) Set(value string) error {
-	var addr *url.URL
-	isURL := a.isURL(value)
-	if isURL {
+	scheme := a.DefaultScheme
+	if len(scheme) == 0 {
+		scheme = "tcp"
+	}
+	addr := &url.URL{
+		Scheme: scheme,
+	}
+
+	switch {
+	case a.isURL(value):
 		parsed, err := url.Parse(value)
 		if err != nil {
 			return fmt.Errorf("not a valid URL: %v", err)
 		}
-		addr = parsed
-	} else {
-		addr = &url.URL{
-			Scheme: a.DefaultScheme,
-			Host:   value,
+		if !a.AllowPrefix {
+			parsed.Path = ""
 		}
-		if len(addr.Scheme) == 0 {
-			addr.Scheme = "tcp"
+		parsed.RawQuery = ""
+		parsed.Fragment = ""
+
+		if strings.Contains(parsed.Host, ":") {
+			host, port, err := net.SplitHostPort(parsed.Host)
+			if err != nil {
+				return fmt.Errorf("not a valid host:port: %v", err)
+			}
+			portNum, err := strconv.ParseUint(port, 10, 64)
+			if err != nil {
+				return fmt.Errorf("not a valid port: %v", err)
+			}
+			a.Host = host
+			a.Port = int(portNum)
+
+		} else {
+			port := 0
+			switch parsed.Scheme {
+			case "http":
+				port = 80
+			case "https":
+				port = 443
+			default:
+				return fmt.Errorf("no port specified")
+			}
+			a.Host = parsed.Host
+			a.Port = port
 		}
-	}
+		addr = parsed
+
+	case isIPv6Host(value):
+		a.Host = value
+		a.Port = a.DefaultPort
 
-	if strings.Contains(addr.Host, ":") {
-		host, port, err := net.SplitHostPort(addr.Host)
+	case strings.Contains(value, ":"):
```

Comment:
No, isIPV6Host still returns true in that case, so you'll never get to this block if you're IPv6